sirepo/pkcli/admin.py

```diff
@@ -24,6 +24,27 @@
 import shutil
 
 
+def add_jupyterhub_role():
+    """Adds jupyterhub role to all users"""
+    import sirepo.server
+
+    sirepo.server.init()
+    import sirepo.auth
+    import sirepo.auth_db
+    import sirepo.pkcli.roles
+    import sirepo.template
+    import sqlalchemy.exc
+
+    sirepo.template.assert_sim_type('jupyterhublogin')
+    r = sirepo.auth.role_for_sim_type('jupyterhublogin')
+    # TODO(e-carlin): locking
+    for u in sirepo.auth_db.all_uids():
+        try:
+            sirepo.pkcli.roles.add(u, r)
+        except sqlalchemy.exc.IntegrityError:
```

Comment:
This shouldn't happen with a db upgrade so let the exception fail. This type of "fail ignore" is problematic, and why I don't particularly like EAFP.

Reply:
Ok. We are now doing the migration when the server is started 252b9c1d.